src/addons/proxymonitor/metaservice/build.gradle

```diff
@@ -13,18 +13,23 @@ sourceSets {
 }
 
 dependencies {
-    compile project(':common-util')
-    compile project(':proxy')
-    compile project(':monitor-common')
-    compile project(':monitoring-conf')
-    compile project(':addons:proxymonitor:common')
-    compile 'com.google.guava:guava:29.0-jre'
-
-    testCompile project (path: ":proxy", configuration: 'testArtifacts')
-    testCompile project(':common-test')
-    testCompile 'org.hamcrest:hamcrest:2.2'
-    testCompile 'com.github.stefanbirkner:system-rules:1.19.0'
-    testCompile "com.typesafe.akka:akka-testkit_$akkaVersion"
+    implementation project(':common-util')
+    implementation project(":serverconf")
+    implementation project(":common-op-monitoring")
+    implementation project(path: ':proxy', configuration: 'shadow')
```

Comment:
Added some comments. Not sure if there is an easy fix for the repeated dependencies. In this particular case, it feels wrong to make proxy a java library (since it is an application) so that its dependencies could be transitively published.